src/util/strencodings.h

```diff
@@ -67,6 +68,66 @@ std::vector<Byte> ParseHex(std::string_view hex_str)
 {
     return TryParseHex<Byte>(hex_str).value_or(std::vector<Byte>{});
 }
+
+/** consteval version of HexDigit() without the lookup table. */
+consteval uint8_t ConstevalHexDigit(const char c)
+{
+    if (c >= '0' && c <= '9') return c - '0';
+    if (c >= 'a' && c <= 'f') return c - 'a' + 0xA;
+
+    throw "Only lowercase hex digits are allowed, for consistency";
+}
+
+struct ConstevalHexLiteral {
+    const std::string_view inner;
+    template <size_t N>
+    consteval ConstevalHexLiteral(const char (&hex_str)[N]) : inner{hex_str} { Validate(inner); }
+    consteval ConstevalHexLiteral(const std::string_view hex_str) : inner{hex_str} { Validate(inner); }
+    consteval ConstevalHexLiteral(std::nullptr_t) = delete;
+
+    static consteval void Validate(const std::string_view hex_str)
+    {
+        std::ranges::for_each(hex_str, ConstevalHexDigit);
+
+        if (hex_str.size() % 2 == 1) throw "2 hex digits required per byte";
```

Comment:
super-nit:
```suggestion
        if (hex_str.size() % 2) throw "2 hex digits required per byte";
```

Reply:
I'm burned by old MSVC warnings (probably not current) "implicit conversion to bool" so I'd rather not.

Reply:
In other cases we're adding and subtracting booleans, but I'm fine with both, see: https://github.com/bitcoin/bitcoin/pull/30535/files#diff-09e6cf871236bf03d32cca9405837d9b7927690b2296a2de17c9be6ea0e75959R74

Reply:
That's some next level terseness. Okay, will change if I re-touch.